README.md

```diff
@@ -28,54 +40,174 @@ A hotel booking application in React. Homework for the [CodeYourFuture React mod
+
+**Hint:** Use `event.target.value` to get the input value.
+
+**Test:** In the developer console, check that everything you type in the search input is printed successively for each new character you enter.
+
+#### 18. Triggering search when submitting the form
+
+**Instructions:** Still in the `<Search />` component, add a `onSubmit` handler to the `<form>` tag. When the form is submitted (try clicking the search button), get the value of the state `searchInput` and pass it as a parameter to the `search` prop function that has been provided for you (the `search` prop is passed from the `<Bookings />` component).
+
+**Note:** Also your submit handler should take an `event` parameter and add the line `event.preventDefault()` to prevent the browser to implicitely submit the form).
+
+**Test:** Look in the console, you should see the text that is typed in the search input field when submitting the form.
+
+#### 19. Implementing the search functionality
+
+**Instructions:** Still in the `<Bookings />` component, implement the `search` method. It must use the `searchVal` variable (that you just passed from the `<Search />` component) to **filter** the search results. The filter function should return bookings where `firstName` or `surname` match `searchVal`. Once filtered, use the `setBookings` function to update the results rendered in `<SearchResults />`.
 
-5. Create a `<SearchResults />` component that shows hotel bookings in a `<table>` element. Each booking will have an id, title, first name, surname, email, room id, check in date and check out date. You can make up data to show in the table. Then show `<SearchResults />` component within the `<Bookings />` component that is provided for you. Be sure to split out your components into small well-named components, similar to the method used in exercise 1. Hint: You will find some useful `<table>` examples in the [Bootstrap documentation for tables](https://getbootstrap.com/docs/4.2/content/tables/#examples)
+**Test:** Verify that when you enter an existing first name or surname and submit the form, the results are filtered accordingly in the customers table.
 
-6. Instead of using your hard-coded data in the `<SearchResults />` component, load data from the `src/data/fakeBookings.json` file in the `<Bookings />` component and pass it as a prop to `<SearchResults />`. Hint: look in the `<Bookings />` component for how to import data from a JSON file.
+#### 20. Highlight only one row when selected
```

Comment:
As discussed above, the best solution to task 16 is to push selection state down to the children. Moving it back up again makes the task easier in some ways.

Reading further, I see why you have done this here though: because it fits well into task 21.

Perhaps we could just add a button to the table row to show in the `CustomerProfile`? Or we could just leave it like this, I don't necessarily think it's a problem per-se.

Reply:
Cool i like the suggestion to add a button, it may make things a bit easier as they won't have to change their highlight implementation from task 16.

Reply:
I've updated task 20 and 21, let me know what you think.